cmd/tpm-v2.c

```diff
@@ -232,6 +232,88 @@ static int do_tpm_get_capability(struct cmd_tbl *cmdtp, int flag, int argc,
 	return report_return_code(rc);
 }
 
+static int do_tpm2_pcrallocate(struct cmd_tbl *cmdtp, int flag, int argc,
+			       char *const argv[])
+{
+	struct udevice *dev;
+	int ret;
+	u16 algo;
+	const char *pw = (argc < 4) ? NULL : argv[3];
+	const ssize_t pw_sz = pw ? strlen(pw) : 0;
+	static struct tpml_pcr_selection pcr = { 0 };
+	u32 pcr_len = 0;
+	bool bon = false;
+	static u32 mask;
+	int i;
+
+	/* argv[1]: algorithm (bank), argv[2]: on/off */
+	if (argc < 3 || argc > 4)
+		return CMD_RET_USAGE;
+
+	if (!strcasecmp("on", argv[2]))
+		bon = true;
+	else if (strcasecmp("off", argv[2]))
+		return CMD_RET_USAGE;
+
+	if (!strcasecmp("TPM2_ALG_SHA1", argv[1])) {
+		algo = TPM2_ALG_SHA1;
+	} else if (!strcasecmp("TPM2_ALG_XOR", argv[1])) {
```

Comment:
U_boot doesnt support this or TPM2_ALG_SM3_256, can we drop them with an error message?